### PR TITLE
Feature/adjusting model

### PR DESCRIPTION
**Pull request by miketillman95**
# Description

Fixes #

- What work was done?
- fixed the add model to add new clients
- Why was this work done?
- What feature / user story is it for?
- Any relevant links or images / screenshots

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [ ] Yes
- [ ] No
- [ ] Not necessary

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: data/migrations/20220117123847_case_mgmt_schema.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 exports.up = function (knex) {
   return knex.schema
     .createTable('locations', (t) => {
-      t.increments('ID').notNullable().unique().primary();
+      t.increments().notNullable().unique().primary();
```
Comment (wild-kard) [inline data/migrations/20220117123847_case_mgmt_schema.js:4]: I'm a bit confused by this change. Shouldn't the ID objects have names?

File: api/client/clientRouter.js [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+router.get('/', (req, res) => {
   Clients.findAll(req.query)
     .then((clients) => {
       res.status(200).json(clients);
     })
     // sending these to the route handlers as well
-    .catch(next);
+    .catch(() => {
+      res.status(500).json({ message: 'internal server error' });
+    });
```
Comment (wild-kard) [inline api/client/clientRouter.js:12]: Good work on the error codes.

File: data/migrations/20220117123847_case_mgmt_schema.js (hunk 1 above)
Comment (miketillman95) [reply data/migrations/20220117123847_case_mgmt_schema.js:4]: 'ID' causes an error in knex when it tries to read the schema. And an  empty`.increments()` populates as an 'id'  in the schema when the code is being run

Comment (wild-kard) [reply data/migrations/20220117123847_case_mgmt_schema.js:4]: Would a more descriptive name like 'location_id" work?

Comment (miketillman95) [reply data/migrations/20220117123847_case_mgmt_schema.js:4]: No the model is looking for the auto increment  of 'id'.

Comment (wild-kard) [reply data/migrations/20220117123847_case_mgmt_schema.js:4]: Makes sense. Are the id's eventually supposed to be UUID's?

**Review by wild-kard — DISMISSED** [review]
Everything is looking okay to me here.

Good work on the error codes you added.

File: data/migrations/20220117123847_case_mgmt_schema.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 exports.up = function (knex) {
   return knex.schema
     .createTable('locations', (t) => {
-      t.increments('ID').notNullable().unique().primary();
```
Comment (wild-kard) [inline data/migrations/20220117123847_case_mgmt_schema.js:4]: I believe you are not supposed to change an existing migration, but rather should update an existing table with a new migration.

Comment (miketillman95) [reply data/migrations/20220117123847_case_mgmt_schema.js:4]: That's one way of doing it. Same end result

Comment (wild-kard) [reply data/migrations/20220117123847_case_mgmt_schema.js:4]: Got it thanks!

File: api/client/clientModel.js [hunk 4]
```diff
@@ -13,7 +13,11 @@ const findAll = () => {
 };
 
 const findById = async (id) => {
-  return db('clients').where({ id }).first();
+  const client = db('clients').where({ id }).first();
```
Comment (stkirk) [inline api/client/clientModel.js:16]: this db call returns a promise, is there a reason you didn't await the response here?

File: api/client/clientModel.js [hunk 5]
```diff
@@ -13,7 +13,11 @@ const findAll = () => {
...
+
+  client.gender = db('genders').where(client.gender_id, 'genders.id').first();
```
Comment (stkirk) [inline api/client/clientModel.js:19]: same concern here, what if the promise doesn't resolve before line 20 returns client?

File: api/client/clientRouter.js [hunk 6]
```diff
@@ -45,10 +51,12 @@ router.delete(':id', (req, res, next) => {
...
       }
     })
-    .catch(next);
+    .catch(() => {
+      res.status(500).json({ message: 'internal server error' });
+    });
 });
 
-router.put(':id', (req, res, next) => {
+router.put(':id', (req, res) => {
```
Comment (stkirk) [inline api/client/clientRouter.js]: you added a "/" before :id on the delete request on line 41, why not here as well?

**Review by stkirk — CHANGES_REQUESTED** [review]

**Review by stkirk — APPROVED** [review]

**Pull request closed without merging**